test/models/user_test.rb

```diff
@@ -320,6 +320,32 @@ def setup_user(operation)
     assert_includes record.errors.keys, :admin
   end
 
+  test "log of password change should be saved" do
+    user = FactoryGirl.create(:user)
+    user.password = nil
+    user.save
+    as_admin do
+      user.password = "newpassword"
+      user.valid?
+      assert user.password_changed_changed?
+      assert user.password_changed
+      assert_includes user.changed, "password_changed"
+    end
+  end
+
+  test "log of password change shouldnt be saved" do
```

Comment:
In this test I want to test that , it should not mark password_changed on every model validation